### PR TITLE
feat: added script copy button

### PR DESCRIPTION
**Pull request by itsarghyadas**
image
Script copy Button added with theme support.

### CODE REVIEW

File: registry/registry-ui.ts [hunk 1]
```diff
@@ -57,6 +57,11 @@ export const ui: Registry = [
     dependencies: ["framer-motion"],
     files: ["magicui/hero-video-dialog.tsx"],
   },
+  {
+    name: "script-copy-btn",
+    type: "registry:ui",
+    files: ["magicui/script-copy-btn.tsx"],
```
Comment (dillionverma) [inline registry/registry-ui.ts:63]: we should list the dependencies here that are being imported by the component

File: registry/default/magicui/dock.tsx [hunk 2]
```diff
@@ -36,12 +36,16 @@ const Dock = React.forwardRef<HTMLDivElement, DockProps>(
...
+      return React.Children.map(children, (child) => {
+        if (React.isValidElement(child) && child.type === DockIcon) {
+          return React.cloneElement(child, {
+            ...child.props,
+            mouseX: mouseX,
+            magnification: magnification,
+            distance: distance,
+          });
+        }
+        return child;
```
Comment (dillionverma) [inline registry/default/magicui/dock.tsx:48]: this dock fix should be in a separate PR ideally just so we know this is not related to the new component

**Review by dillionverma — CHANGES_REQUESTED** [review]
Some minor fixes

Also, you should run `pnpm format:write` and `pnpm lint:fix` locally to clean the code before pushing up

File: registry/default/magicui/script-copy-btn.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
+"use client";
+
+import { useState, useEffect } from "react";
+import { Button } from "@/components/ui/button";
+import { Check, Copy } from "lucide-react";
+import { codeToHtml } from "shiki";
+import { motion } from "framer-motion";
+import { useTheme } from "next-themes";
```
Comment (dillionverma) [inline registry/default/magicui/script-copy-btn.tsx:8]: what should be done for non next.js users?

Comment (itsarghyadas) [reply registry/default/magicui/script-copy-btn.tsx:8]: It will work for non-next.js users too.
